src/Frontend/Core/Engine/Model.php

```diff
@@ -326,6 +329,29 @@ public static function getModuleSettings($module)
     }
 
     /**
+     * Creates router for module based on routing file
+     *
+     * @param $module
+     * @return null|Router
+     */
+    public static function getModuleRouter($module)
```

Comment:
If we could pass Module object through methods instead of just module name, that would solve lots of problems. But this is huge work i think.